components/camel-spring-rabbitmq/src/main/java/org/apache/camel/component/springrabbit/DefaultMessagePropertiesConverter.java

```diff
@@ -39,9 +40,64 @@ public DefaultMessagePropertiesConverter(CamelContext camelContext, HeaderFilter
     @Override
     public MessageProperties toMessageProperties(Exchange exchange) {
         MessageProperties answer = new MessageProperties();
-        String contentType = ExchangeHelper.getContentType(exchange);
+
+        Object deliveryMode = exchange.getMessage().removeHeader(SpringRabbitMQConstants.DELIVERY_MODE);
+        if (deliveryMode != null) {
+            answer.setDeliveryMode(deliveryMode instanceof MessageDeliveryMode ? (MessageDeliveryMode) deliveryMode
+                    : MessageDeliveryMode.fromInt(Integer.parseInt(deliveryMode.toString())));
+        }
+        Object type = exchange.getMessage().removeHeader(SpringRabbitMQConstants.TYPE);
```

Comment:
Is-it necessary to remove the header?

Reply:
Yes we do not want it to be copied later on to the RabbitMQ message.

Reply:
on second thought maybe this is something that header filter strategy should take care of? we may want to re-use a header or two after the message has been published.

Reply:
What I can tell for sure is that it is quite unexpected that a converter alters the message. If they need to be removed it should be definitively done elsewhere IMHO